packages/agents/src/index.ts

```diff
@@ -776,6 +908,80 @@ export class Agent<Env, State = unknown> extends Server<Env> {
     // biome-ignore lint/complexity/noBannedTypes: <explanation>
     return callableMetadata.has(this[method as keyof this] as Function);
   }
+
+  /**
+   * Connect to a new MCP Server
+   *
+   * @param url MCP Server SSE URL
+   * @param callbackHost Base host for the agent, used for the redirect URI.
+   * @param agentsPrefix agents routing prefix if not using `agents`
+   * @returns authUrl
+   */
+  async addMcpServer(
+    url: string,
+    callbackHost: string,
+    agentsPrefix = "agents"
+  ): Promise<{ id: string; authUrl: string | undefined }> {
```

Comment:
i think it could be useful to be able to give the server a name. i kind of had that use case because i wanted to dynamically show something like `Please authenticate with ${serviceName}` or `Connected to ${serviceName}`. i could keep a mapping of url -> name/metadata somewhere, but meh...

Reply:
Yeah good point, I might add that. 

I wish I could grab the serverName from the initialization handshake, but unfortunately that requires authorizing beforehand, which wouldn't solve your case: https://modelcontextprotocol.io/specification/2025-03-26/basic/lifecycle#initialization